fmperf/loadgen/run.py

```diff
@@ -127,26 +127,28 @@ def worker(wid, channel):
                     else:
                         apply_backoff = True
 
-                record = {
-                    "response": r,
-                    "ok": ok,
-                    "error": str(err),
-                    "timestamp": t,
-                    "exp_num_users": num_users,
-                    "exp_duration": duration.to_seconds(),
-                    "duration_ms": (t - t0) / 1000.0 / 1000.0,
-                    "exclude": (t - t_start) / 1000.0 / 1000.0 / 1000.0
-                    > (duration.to_seconds() + grace_period.to_seconds()),
-                    "worker_idx": wid,
-                    "request_idx": request_idx,
-                    "sample_idx": sample_idx,
-                    "response_idx": response_idx,
-                    "n_tokens": n_tokens,
-                }
-
-                output.append(record)
-                response_idx += 1
-                t0 = t
+                # ignore 'empty' tokens at the beginning of the output (can occur because of chunked prefills)
+                if not ((r['text'] == '') and (response_idx == 0)):
```

Comment:
Is there any way this logic could be implemented inside `get_streaming_response_vllm` to reflect that it is a unique behaviour to vLLM, rather that both vLLM and TGIS? Not such a big deal if not.

Reply:
Sure, np. Simply including an if-statement (see below - "filter empty tokens") in  `get_streaming_response_vllm` does the job:
```
    def get_streaming_response_vllm(response):
        response_iter = response.iter_lines(
            chunk_size=8192,
            decode_unicode=False,
            delimiter=b"\n",
        )

        stop = False
        while not stop:
            try:
                chunk = next(response_iter)
                timestamp = time.time_ns()
                if chunk and not stop:
                    data = chunk.decode("utf-8").strip().split("data: ")[1]
                    out = json.loads(data)["choices"][0]
                    stop = out["finish_reason"] is not None
                    if not (out['text'] == ''): # filter empty tokens
                        yield out, 1, timestamp, True, None
            except Exception as e:
                timestamp = time.time_ns()
                yield None, 0, timestamp, False, e

        # we have stopped
        yield None, 0, time.time_ns(), False, StopIteration()
```

Reply:
In this case, all "empty-tokens" are filtered out. If we want to explicitly filter out "empty-tokens" only at the beginning of a sequence, I need to add an additional variable to keep track if we are at the beginning or not. Just let me know what you prefer.

I will run some further experiments, with this code, and the commit it, if it works fine.

Reply:
For consistency, I will make a similar update to `get_streaming_response()` in `generate-input.py`. Same question as above: do we want to check explicitly for removing "empty tokens" at the beginning of an output sequence or do we remove "empty tokens" at any position.

Reply:
It LGTM